geoips/interfaces/base.py

```diff
@@ -0,0 +1,360 @@
+            )
+        else:
+            raise PluginError(
+                f"Required '__doc__' attribute not found in '{module.__name__}'"
+            )
+
+    try:
+        obj_attrs["name"] = module.name
+    except AttributeError:
+        if plugin_deprecations:
+            warn(
+                f"Variable 'name' not found in plugin module '{module.__name__}'."
+                f"Assuming module name is "
+                f"`{'.'.join(module.__name__.split('.')[-2:])}`. "
+                "Please update to add 'name' variable."
+            )
+            obj_attrs["name"] = module.__name__.split(".")[-1]
+        else:
+            raise PluginError(
+                f"Required 'name' attribute not found in '{module.__name__}'"
+            )
+
+    try:
+        obj_attrs["family"] = module.family
+    except AttributeError:
+        if plugin_deprecations:
+            warn(
+                f"Use of the '{interface.deprecated_family_attr}' attribute is "
+                "deprecated. All uses should be replaced with an attribute named "
+                "'family'. In the future this will result in a TypeError.",
+                DeprecationWarning,
+                stacklevel=1,
+            )
+        else:
+            raise PluginError(
+                f"Required 'family' attribute not found in '{module.__name__}'"
+            )
+        try:
+            obj_attrs["family"] = getattr(module, interface.deprecated_family_attr)
+        except AttributeError:
+            raise PluginError(
+                f"Required 'family' attribute not found in '{module.__name__}'"
+            )
+
+    try:
+        obj_attrs["description"] = module.description
+    except AttributeError:
+        if plugin_deprecations:
+            warn(
+                f"Plugin module '{module.__name__}' does not implement a 'description' "
+                "attribute. In future releases this will result in a TypeError.",
+                DeprecationWarning,
+                stacklevel=1,
+            )
+            obj_attrs["description"] = None
+        else:
+            raise PluginError(
+                f"Required 'description' attribute not found in '{module.__name__}'"
+            )
+
+    if module_call_func != "call":
+        warn(
+            f"Callable for plugin '{module.__name__}' is not named 'call'. "
+            "This behavior is deprecated. The callable should be renamed to "
+            "'call'. In the future this will result in a TypeError.",
+            DeprecationWarning,
+            stacklevel=1,
+        )
+    # Collect the callable and assign to __call__
+    try:
+        obj_attrs["__call__"] = staticmethod(getattr(module, module_call_func))
+    except AttributeError as err:
+        raise PluginError(
+            f"Plugin module '{module.__name__}' does not implement a function named "
+            f"'{module_call_func}'."
+        ) from err
+
+    plugin_type = f"{interface.name.title()}Plugin"
```

Comment:
@jsolbrig Oh - here is where it becomes AlgorithmInterfacePlugin (RE my comment on interface_notes.md)

Reply:
Yeah, these classes are currently automatically generated but I would like to clean them up some as mentioned in #75.